### PR TITLE
Add arrow key function to text inputs

### PR DESCRIPTION
**Pull request by BlackWolfie**
- Extend key down event listener on text input
- When the arrow key down/up is pressed:
    * Check if value is numeric CSS unit
    * Increase pixel/percent values by 1
    * Increase numeric/em/rem/vw/vh values by 0.1
    * with `ctrl` + arrow pressed:
        * Increase pixel/percent values by 10
        * increase numeric/em/rem/vw/vh values by 1

### CODE REVIEW

File: editor/src/app/inputs/text-input.service.ts [hunk 1]
```diff
@@ -44,16 +44,56 @@ export class TextInputService {
...
     }
 
+    if (event.key === 'ArrowDown' || event.keyCode === 40 || event.key === 'ArrowUp' || event.keyCode === 38) {
+      const myLabel = ['Page margins', 'Margins', 'Gallery margins', 'Entry margins'];
+      const bigUnits = ['px', '%', 'pt'];
+      const litleUnits = ['em', 'rem', 'vw', 'vh'];
+
+      if (!myLabel.includes(label)) {
+        const regex  = /[a-z%]+/g;
+        const regex1 = /-?[0-9]\d*(\.\d+)?/g;
```
Comment (vteivans) [inline editor/src/app/inputs/text-input.service.ts]: why are your regexes called `regex` and `regex1`? what are they for?

File: editor/src/app/inputs/text-input.service.ts [hunk 2]
```diff
@@ -44,16 +44,56 @@ export class TextInputService {
...
       (event.target as HTMLInputElement).blur();
       return null;
     }
 
+    if (event.key === 'ArrowDown' || event.keyCode === 40 || event.key === 'ArrowUp' || event.keyCode === 38) {
+      const myLabel = ['Page margins', 'Margins', 'Gallery margins', 'Entry margins'];
+      const bigUnits = ['px', '%', 'pt'];
+      const litleUnits = ['em', 'rem', 'vw', 'vh'];
+
+      if (!myLabel.includes(label)) {
```
Comment (vteivans) [inline editor/src/app/inputs/text-input.service.ts]: This is a very weird implementation. If you wish to filter out specific settings that will have this feature you should do it before - when the inputs are created, you can add a parameter, that determines if they will have this arrow key feature.

That should most likely be determined by the type of the setting input.

File: editor/src/app/inputs/text-input.service.ts (hunk 1 above)
Comment (BlackWolfie) [reply editor/src/app/inputs/text-input.service.ts]: one regexe to get the unit regex;  and one for the negative and decimal number regex1
i have to rename it i think

File: editor/src/app/inputs/text-input.service.ts [hunk 3]
```diff
@@ -48,12 +48,50 @@ export class TextInputService {
...
+      const bigUnits = ['px', '%', 'pt'];
+      const litleUnits = ['em', 'rem', 'vw', 'vh'];
+
+      const unitRegex  = /[a-z%]+/g;
+      const digitRegex = /-?[0-9]\d*(\.\d+)?/g;
+
+      const found = event.target.value.match(unitRegex);
+      const digit = event.target.value.match(digitRegex);
+
+      if (digit !== null && digit.length <= 1 ) {
```
Comment (uldisrudzitis) [inline editor/src/app/inputs/text-input.service.ts]: You should also check if given unit is in `bigUnits` or `litleUnits`. If there is wrong unit given `10pux` result is `NaNpux` when arrow key is pressed.

File: editor/src/app/inputs/text-input.service.ts [hunk 4]
```diff
@@ -48,12 +48,50 @@ export class TextInputService {
...
+
     if (event instanceof KeyboardEvent && (event.key === 'Escape' || event.keyCode === 27)) {
       (event.target as HTMLInputElement).value = this.lastValue;
       (event.target as HTMLInputElement).blur();
       return null;
     }
 
+    if (event.key === 'ArrowDown' || event.keyCode === 40 || event.key === 'ArrowUp' || event.keyCode === 38) {
+      const bigUnits = ['px', '%', 'pt'];
+      const litleUnits = ['em', 'rem', 'vw', 'vh'];
```
Comment (uldisrudzitis) [inline editor/src/app/inputs/text-input.service.ts]: `litleUnits -> littleUnits`
or better rename them to `integerUnits`, `decimalUnits`

File: editor/src/app/inputs/text-input.service.ts [hunk 5]
```diff
@@ -48,12 +48,50 @@ export class TextInputService {
...
+        }
+
+        if (event.key === 'ArrowDown' || event.keyCode === 40) {
+          value = Math.round((value - i) * 10) / 10;
+        }
+
+        if (event.key === 'ArrowUp' || event.keyCode === 38) {
+          value = Math.round((value + i) * 10) / 10;
+        }
+        (event.target as HTMLInputElement).value = value + unit; // fonction pour changer la value
```
[the remark this answers is not in the document]
Comment (BlackWolfie) [reply editor/src/app/inputs/text-input.service.ts]: sry my bad

**Pull request merged**